core/bentley/src/TupleKeyedMap.ts

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Collections
+ */
+
+/** A map similar to the standard JavaScript Map collection except that the keys must be a tuple
+ * (javascript array), and value comparison is used on these tuple keys, without a user-provided hash function.
```

Comment:
Reference to "user-provided hash function" is confusing - how is it relevant? JavaScript maps don't use user-defined hash functions either.

"value comparison" is also confusing. Should you be clearer and state that it uses the same [SameValueZero](https://262.ecma-international.org/6.0/#sec-samevaluezero) comparison as Map on each element of the tuples?

Reply:
I think the but the best known concept for js devs is [strict equality](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality), so I can say something about it considering two tuples equal if their elements in order at strict-equal.

Referencing the hash function stuff was me acknowledging that some people would do this without nested maps and instead by mapping keys to strings and using those strings as keys in the value-compared map. I will remove that since it's unrelated to what this map does

Reply:
But Maps don't use strict equality. e.g., NaN, -0, etc. You're giving them a variation of a Map.

Reply:
you're right I was unaware of that distinction. I will add a reference to the standard

Reply:
updated docs with better desc and ref to standard about the distinction